include/sound/sof/ipc4/header.h

```diff
@@ -0,0 +1,333 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
+/*
```

Comment:
the commit message looks a lot easier to read @bardliao But just wondering is it important to mention closed-source here?

Reply:
closed source ? I prefer windows audio FW since now we are Linux audio FW. Open or close, it is topic for debate.

Reply:
@ranj063 @RanderWang The commit message explains why we need IPC4. To be compatible with the existing firmware is an important reason of using IPC4. So yes, I think it is important to mention here. But I don't have any idea/opinion what should we call the existing firmware. I will follow the majority.

Reply:
sorry @bardliao i dont quite like both closed-source and Windows in the commit message. How about we just say that the SOF FW supports a new IPC definition called IPC4 and this patch adds the kernel support for it?

Reply:
@ranj063. they will ask you why introduce new IPC definition and is it better than current ipc ?  it is not better than current one, only to reuse current FW.

Reply:
@RanderWang, I would not compare this to NTFS, you can 'use' it in Linux, yes.

The issue is that we are implying that IPC4 == closed firmware, which is far from the truth. IPC4 should be a communication protocol, an open standard which is implemented by a closed firmware as well. It is more like XMPP.

For me we are also mixing the definition of what IPC4 is supposed to be and tying it to a given firmware.

Reply:
@ujfalusi  IPC3 is enough for open source and we will go on supporting it. The major reason for IPC4 is to reuse current fw, not to give another solution for open source. yes, it is better not to bind it to a given firmware and it will also support open source.

Reply:
> sorry @bardliao i dont quite like both closed-source and Windows in the commit message. How about we just say that the SOF FW supports a new IPC definition called IPC4 and this patch adds the kernel support for it?

@ranj063 Sounds good. Simpler is better. I updated the commit message as you suggested on #3149

Reply:
@bardliao @RanderWang @ujfalusi @ranj063 To clarify, IPC3 and IPC4 will both be supported going forward until we can align on a more dynamic IPC (like IPC tuples). IPC4 is NOT to support closed source legacy FW, but is to support SOF on Windows in the immediate future before we can align on IPC tuples at a later date.